src/Templates/AdminPanel/Bit.AdminPanel/Web/Extensions/HttpRequestExtensions.cs

```diff
@@ -0,0 +1,52 @@
+﻿using Microsoft.Net.Http.Headers;
+
+namespace Microsoft.AspNetCore.Http;
+
+public static class HttpRequestExtensions
+{
+    public static bool IsSearchEngineBot(this HttpRequest request)
+    {
+        var agent = GetUserAgent(request);
+
+        var isGoogleBot = IsGoogleBot(agent);
```

Comment:
there is no need to define these simple methods and complicate code like this.
plz remove these methods and directly check the user-agent here.